src/network/networkd-link.c

```diff
@@ -99,6 +99,21 @@ static bool link_ipv6ll_enabled(Link *link) {
         return link->network->link_local & ADDRESS_FAMILY_IPV6;
 }
 
+static bool link_ipv6_enabled(Link *link) {
+        assert(link);
+
+        if (!socket_ipv6_is_supported())
+                return false;
+
+        if (link_dhcp6_enabled(link) && link_ipv6ll_enabled(link))
+                return true;
```

Comment:
Errr, no. If I have DHCP=ipv6 LinkLocalAddressing=ipv4 I'm pretty sure this should return true.